### PR TITLE
Tweak: Refactor - separate overview widget to sections [ED-9571]

### PR DESCRIPTION
**Pull request by lenalElementor**

### CODE REVIEW

File: core/admin/admin.php [hunk 1]
```diff
@@ -410,71 +435,140 @@ public function elementor_dashboard_overview_widget() {
...
+	 *
+	 * @since 1.9.0
+	 * @access public
+	 *
+	 * @param array $args
+	 * @param bool $show_heading
+	 *
+	 * @return void
+	 */
+	public function elementor_dashboard_overview_recently_edited( $args = [], $show_heading = true ) {
```
Comment (nivnoimanelementor) [inline core/admin/admin.php:480]: Don't use show_heading. Instead split it into another function that can be invoke.

**Review by nivnoimanelementor — CHANGES_REQUESTED** [review]
Please check the request changes,

File: core/admin/admin.php [hunk 2]
```diff
@@ -410,71 +435,140 @@ public function elementor_dashboard_overview_widget() {
...
+
+	/**
+	 * Elementor dashboard widget.
+	 *
+	 * Displays the Elementor dashboard widget - recently edited section.
+	 *
+	 * Fired by `elementor_dashboard_overview_widget` function.
+	 *
+	 * @since 3.12.0
+	 * @access public
```
Comment (mark-elementor) [inline core/admin/admin.php:473]:
```suggestion
```

File: core/admin/admin.php [hunk 3]
```diff
@@ -410,71 +435,140 @@ public function elementor_dashboard_overview_widget() {
...
+	 *
+	 * @param array $args
+	 * @param bool $show_heading
+	 *
+	 * @return void
+	 */
+	public function elementor_dashboard_overview_recently_edited( $args = [], $show_heading = true ) {
```
Comment (mark-elementor) [inline core/admin/admin.php:480]:
```suggestion
	public function elementor_dashboard_overview_recently_edited( array $args = [], bool $show_heading = true ) {
```
Also, maybe "is_heading_shown" or something like that?

File: core/admin/admin.php [hunk 4]
```diff
@@ -410,71 +435,140 @@ public function elementor_dashboard_overview_widget() {
...
-						<li class="e-overview__<?php echo esc_attr( $action_id ); ?>"><a href="<?php echo esc_attr( $action['link'] ); ?>" target="_blank"><?php echo esc_html( $action['title'] ); ?> <span class="screen-reader-text"><?php echo esc_html__( '(opens in a new window)', 'elementor' ); ?></span><span aria-hidden="true" class="dashicons dashicons-external"></span></a></li>
-					<?php endforeach; ?>
+			<?php } ?>
+		</div>
+		<?php
+	}
+
+	/**
+	 * Elementor dashboard widget.
+	 *
```
Comment (mark-elementor) [inline core/admin/admin.php:467]:
```suggestion
```

File: core/admin/admin.php [hunk 5]
```diff
@@ -410,71 +435,140 @@ public function elementor_dashboard_overview_widget() {
...
+	 * @since 3.12.0
+	 * @access public
+	 *
+	 * @param int $limit_feed
+	 * @param bool $show_heading
+	 *
+	 * @return void
+	 */
+	public function elementor_dashboard_overview_news_updates( $limit_feed = 0, $show_heading = true ) {
```
Comment (mark-elementor) [inline core/admin/admin.php:522]:
```suggestion
	public function elementor_dashboard_overview_news_updates( int $limit_feed = 0, bool $show_heading = true ) {
```
$show_heading naming?

File: core/admin/admin.php [hunk 6]
```diff
@@ -410,71 +435,140 @@ public function elementor_dashboard_overview_widget() {
...
+
+	/**
+	 * Elementor dashboard widget.
+	 *
+	 * Displays the Elementor dashboard widget - news and updates section.
+	 *
+	 * Fired by `elementor_dashboard_overview_widget` function.
+	 *
+	 * @since 3.12.0
+	 * @access public
```
Comment (mark-elementor) [inline core/admin/admin.php:515]:
```suggestion
```

File: core/admin/admin.php [hunk 7]
```diff
@@ -410,71 +435,140 @@ public function elementor_dashboard_overview_widget() {
...
+						</li>
+					<?php } ?>
 				</ul>
 			</div>
+		<?php }
+	}
+
+	/**
+	 * Elementor dashboard widget.
+	 *
```
Comment (mark-elementor) [inline core/admin/admin.php:509]:
```suggestion
```

File: core/admin/admin.php [hunk 8]
```diff
@@ -410,71 +435,140 @@ public function elementor_dashboard_overview_widget() {
...
+						</li>
+					<?php } ?>
+				</ul>
+			</div>
+		<?php }
+	}
+
+	/**
+	 * Elementor dashboard widget.
+	 *
```
Comment (mark-elementor) [inline core/admin/admin.php:552]:
```suggestion
```

File: core/admin/admin.php [hunk 9]
```diff
@@ -410,71 +435,140 @@ public function elementor_dashboard_overview_widget() {
...
+	 * Displays the Elementor dashboard widget - footer section.
+	 *
+	 * Fired by `elementor_dashboard_overview_widget` function.
+	 *
+	 * @since 3.12.0
+	 * @access public
```
Comment (mark-elementor) [inline core/admin/admin.php:558]:
```suggestion
```

**Review by mark-elementor — CHANGES_REQUESTED** [review]

**Comment by mark-elementor** [comment]
General comment - maybe should move the dashboard widgets to a separate class?

File: core/admin/admin.php [hunk 10]
```diff
@@ -399,9 +399,34 @@ public function register_dashboard_widgets() {
...
+	 *
+	 * @param bool $show_versions
+	 * @param bool $show_create_post
+	 *
+	 * @return void
+	 */
+	public function elementor_dashboard_overview_header( $show_versions = true, $show_create_post = true ) {
```
[the remark this answers is not in the document]
Comment (lenalElementor) [reply core/admin/admin.php:429]: I don’t think we support param type declaration.

File: core/admin/admin.php [hunk 11]
```diff
@@ -399,9 +399,34 @@ public function register_dashboard_widgets() {
...
 
+	/**
+	 * Elementor dashboard widget.
+	 *
+	 * Displays the Elementor dashboard widget - header section.
+	 *
+	 * Fired by `elementor_dashboard_overview_widget` function.
+	 *
+	 * @since 3.12.0
+	 * @access public
```
[the remark this answers is not in the document]
Comment (lenalElementor) [reply core/admin/admin.php:422]: Why remove this line?

File: core/admin/admin.php [hunk 12]
```diff
@@ -399,9 +399,34 @@ public function register_dashboard_widgets() {
...
+			$this->elementor_dashboard_overview_news_updates();
+			$this->elementor_dashboard_overview_footer();
+			?>
+		</div>
+		<?php
+	}
 
+	/**
+	 * Elementor dashboard widget.
+	 *
```
[the remark this answers is not in the document]
Comment (lenalElementor) [reply core/admin/admin.php:416]: Why?

File: core/admin/admin.php [hunk 13]
```diff
@@ -505,21 +492,17 @@ public function elementor_dashboard_overview_recently_edited( $args = [], $show_
...
 	 *
-	 * @since 3.12.0
-	 * @access public
-	 *
 	 * @param int $limit_feed
 	 * @param bool $show_heading
 	 *
 	 * @return void
+	 *@since 3.12.0
+	 * @access public
```
Comment (mark-elementor) [inline core/admin/admin.php:503]: Something weird happent here.

**Review by mark-elementor — DISMISSED** [review]

File: assets/dev/scss/admin/_dashboard-widget.scss [hunk 14]
```diff
@@ -160,3 +160,7 @@
 		}
 	}
 }
+
+.e-dashboard-overview {
```
Comment (nivnoimanelementor) [inline assets/dev/scss/admin/_dashboard-widget.scss:164]: why we still have all the style in the class ID as well ?
why not to replace the `#` to `.` at the original ?

**Review by nivnoimanelementor — CHANGES_REQUESTED** [review]
please check this last thing 🙏

**Review by nivnoimanelementor — APPROVED** [review]

**Pull request merged**